### PR TITLE
Developer documentation update

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: doc/intro.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+BlivetGUI
+^^^^^^^^^
+
+:class:`~.blivetgui.BlivetGUI` is the main class for the GUI. There are separate classes for every UI part. These classes
+creates all the necessary Gtk widgets (or loads them from Glade files) and provides signal handlers and other helper
+functions.
+
+Important GUI objects:
+ - :class:`~.actions_menu.ActionsMenu` -- context (left click) menu for user actions
```
Comment (vpodzime) [inline doc/intro.rst]: Is this really a left-click context menu? Or should it be *right click*?

File: doc/intro.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ - :class:`~.actions_toolbar.DeviceToolbar` -- toolbar with device-related actions (add, edit, remove...)
+ - :class:`~.actions_toolbar.ActionsToolbar` -- toolbar with "global" actions (process or clear actions, main menu)
+ - :class:`~.list_devices.ListDevices` -- list of "root" devices (disks, volume groups, RAIDs...)
+ - :class:`~.visualization.logical_view.LogicalView` -- displays list of children of selected device (both as list using `Gtk.TreeView` and graphically using :class:`~.visualization.rectangle.Rectangle`)
+ - :class:`~.visualization.physical_view.PhysicalView` -- displays list of parents of selected device
+
+Dialogs
+^^^^^^^
+
+Most of the user interaction are done using dialogs. Some important dialogs include:
```
Comment (vpodzime) [inline doc/intro.rst]: *is* done...

File: doc/intro.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+Blivet-gui uses `pickle` to send objects between the processes. Unfortunately most blivet objects are not picklable. To solve this, server and
+client parts of blivet-gui communicate using "proxy objects".
+When :class:`~.blivetgui.BlivetGUI` "asks" for some "information" that isn't picklable (e.g. using :meth:`~.blivet_utils.BlivetUtils.get_disks`
+that returns list of :class:`blivet.devices.storage.StorageDevice`), :class:`~.communication.server.BlivetUtilsServer` creates a new instance
+of :class:`~.communication.proxy_utils.ProxyID` and sends it to the client instead. When client recieves it, it creates an instance of
+:class:`~.communication.client.ClientProxyObject` and sends it to the :class:`~.blivetgui.BlivetGUI` instead.
+
+:class:`~.communication.client.ClientProxyObject` has a custom `__getattr__` method and has the ProxyID so it can ask the server for values
+of attributes that :class:`~.blivetgui.BlivetGUI` asks for. Thanks to this, :class:`~.blivetgui.BlivetGUI` can work with the blivet object
+without having it.
```
Comment (vpodzime) [inline doc/intro.rst]: Might be nice to wrap these long lines.

File: doc/intro.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+::
+
+  Note: It isn't possible to call methods of proxied objects. Only attributes (and properties) are supported.
```
Comment (vpodzime) [inline doc/intro.rst]: ReST has the ``.. note::`` admonition.

**Comment by vpodzime** [comment]
Looks good to me otherwise.

**Comment by vojtechtrefny** [comment]
New version based on @vpodzime comments.

**Pull request merged**